### PR TITLE
feature test for guest user cart closes #15

### LINKED ISSUE #15: guest user

### PR DESCRIPTION
**Pull request by mariastenquist**
-wrote feature test cart persists when a guest user logs in
-a guest user cannot checkout unless they are logged in

### CODE REVIEW

**Comment by samlandfried** [comment]
@dannyradden @mariastenquist this branch is ready for review - this is a branch off of the pull request for 14

File: app/views/carts/show.html.erb [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+    <br>
+    <%= button_to "Remove Item", cart_path(creature_id: creature.id), method: :delete %>
+  </li>
   <% end %>
 </ul>
 <div class="cart-total">
   Total Price: <%= number_to_currency @cart.total_price %>
 </div>
+<% if current_user %>
+<%= button_to 'Checkout', '/orders/new', class: 'checkout-btn' %>
```
Comment (jk1dd) [inline app/views/carts/show.html.erb]: fix indentation @dannyradden

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -21,13 +21,20 @@
         <ul id="nav-mobile" class="right hide-on-med-and-down">
           <li><a href="sass.html">Place Holder 1</a></li>
           <li><a href="badges.html">Place Holder 2</a></li>
+          <% if current_user %>
+            <li>Logged in as <%= current_user.username %></li>
```
Comment (jk1dd) [inline app/views/layouts/application.html.erb]: let's make this username a link to that user's dashboard @samlandfried

File: spec/features/cart_persists_when_guest_user_logs_in_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.feature 'Guest user cart functionality' do
+  scenario 'a guest user needs to login before checkout when items in cart' do
+    # As a visitor when I have items in my cart
```
Comment (jk1dd) [inline spec/features/cart_persists_when_guest_user_logs_in_spec.rb]: let's delete these commented lines @mariastenquist

**Pull request merged**